src/backend/distributed/transaction/transaction_management.c

```diff
@@ -194,7 +196,17 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			 * transaction management. Do so before doing other work, so the
 			 * callbacks still can perform work if needed.
 			 */
-			RemoveIntermediateResultsDirectory();
+			{
+				/*
+				 * On Windows it's not possible to delete a file before you've closed all
+				 * handles to it (rmdir will return success but not take effect). Since
+				 * we're in an ABORT handler it's very likely that not all handles have
+				 * been closed; force them closed here before running
+				 * RemoveIntermediateResultsDirectory.
+				 */
+				AtEOXact_Files();
```

Comment:
Could there be any ERRORs in here?

Reply:
Do you mean in the call to `AtEOXact_Files()`? I don't think so, it's called by Postgres just after calling the user xact callbacks, if I remember correctly.